Source/utils/lua.cpp

```diff
@@ -0,0 +1,57 @@
+#include "utils/lua.hpp"
+
+extern "C" {
+#include <lauxlib.h>
+#include <lua.h>
+#include <lualib.h>
+}
+
+#include "plrmsg.h"
+
+namespace devilution {
+
+namespace {
+
+lua_State *LuaState;
+
+int LuaPlayerMessage(lua_State *state)
+{
+	if (lua_isstring(LuaState, 1)) {
+		std::string msg = lua_tostring(state, 1);
+		EventPlrMsg(msg, UiFlags::ColorRed);
+	}
+
+	return 0;
+}
+
+} // namespace
+
+void LuaInitialize()
+{
+	LuaState = luaL_newstate();
+	lua_register(LuaState, "PlayerMessage", LuaPlayerMessage);
+
+	int status = luaL_dofile(LuaState, "init.lua");
+	if (status != LUA_OK)
+		SDL_Log("%s", lua_tostring(LuaState, -1));
+}

```

Comment:
You'll want to hook Lua's `print()` up to the logger, I'm not sure if there's a mechanic that's offered but the way I've seen another game do it is to redefine the built-in print.